### PR TITLE
Python: add missing ai_model_id in ollama request

### PR DESCRIPTION
**Pull request by imfht**
### Motivation and Context
I installed `semantic-kernel==0.4.5.dev0` and `ollama 0.1.20e`. and want to test sk with ollama. 
and got error when i write a sample code.
```
Error description: '400, message='Bad Request', url=URL('http://192.168.11.2:11434/api/chat')'
```
I dived in to the call stack and found it's caused by missing ai_model_id, so I make some little changes and it works well.
here is current version with bug, we can see the model id is empty. 
image
after fix, it works well
image

it is related to #4527 .

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by imfht** [comment]
## 1. Ollama document
Ollama document shows models is required.
> model: (required) the [model name](https://github.com/jmorganca/ollama/blob/main/docs/api.md#model-names)
messages: the messages of the chat, this can be used to keep a chat memory

https://github.com/jmorganca/ollama/blob/main/docs/api.md#parameters-1

## 2. this code can produce (need ollama installed on locahost)
```python
import semantic_kernel as sk
from semantic_kernel.connectors.ai.ollama.services.ollama_chat_completion import OllamaChatCompletion

from config import kernel

kernel = sk.Kernel()

kernel.add_chat_service(
    "llama2",
    OllamaChatCompletion(ai_model_id="llama2", url='http://127.0.0.1:11434/api/chat'),
)


async def main():
    import semantic_kernel as sk
    from semantic_kernel.core_skills import ConversationSummarySkill
    from semantic_kernel.core_skills import TimeSkill
    # import config.add_completion_service

    # Initialize the kernel
    # Add a text or chat completion service using either:
    # kernel.add_text_completion_service()
    # kernel.add_chat_service()

    # Import the ConversationSummaryPlugin
    kernel.import_skill(
        ConversationSummarySkill(kernel=kernel), skill_name="ConversationSummaryPlugin"
    )
    kernel.import_skill(TimeSkill(), skill_name="time")

    # Create the history
    history = []

    # Create the prompt with the ConversationSummaryPlugin
    prompt = """{{ConversationSummaryPlugin.SummarizeConversation $history}}
    today is: {{time.now}}
    User: {{$request}}
    Assistant:  """

    while True:
        request = input("User > ")

        variables = sk.ContextVariables()
        variables["request"] = request
        variables["history"] = "\n".join(history)

        # Run the prompt
        semantic_function = kernel.create_semantic_function(prompt)
        result = await kernel.run_async(
            semantic_function,
            input_vars=variables,
        )

        # Add the request to the history
        history.append("User: " + request)
        history.append("Assistant" + result.result)

        print("Assistant > " + result.result)


# Run the main function
if __name__ == "__main__":
    import asyncio

    asyncio.run(main())

```

**Comment by imfht** [comment]
Hi @moonbox3  cloud you have a look at this?

**Review by juliomenendez — APPROVED** [review]

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 1]
```diff
@@ -30,11 +30,11 @@ class OllamaChatCompletion(TextCompletionClientBase, ChatCompletionClientBase, A
     Make sure to have the ollama service running either locally or remotely.
 
     Arguments:
-        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library, defaults to llama2
         url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/chat
         session {Optional[aiohttp.ClientSession]} -- Optional client session to use for requests.
     """
-
+    ai_model_id: str = "llama2"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:37]: This field is already part of AI client base, and has important annotation. Should be overwritten if necessary with those.

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
See comment

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:37]: I see what @eavanvalkenburg is pointing out. You'll want to follow how it's done in another example, like [OpenAI chat completion](https://github.com/microsoft/semantic-kernel/blob/e4874bb79d7c4f50c6975353ca9032f2982d4a60/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py#L177C9-L178C52).

Comment (moonbox3) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:37]: I am seeing in this example that before the ollama call, the `request_settings.ai_model_id = ''` which means it's just empty and not None.

**Comment by imfht** [comment]
> ```
> @microsoft-github-policy-service agree
> ```

@microsoft-github-policy-service agree

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py (hunk 1 above)
Comment (imfht) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:38]: new commit added, I use same code style with [OpenAI chat completion](https://github.com/microsoft/semantic-kernel/blob/e4874bb79d7c4f50c6975353ca9032f2982d4a60/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py#L177C9-L178C52).

**Comment by imfht** [comment]
> Hi @imfht, thank you for your help on this. Could you run `black` formatting, please? The change looks good to me, so will approve once the formatting is fixed. Thanks.

done

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ async def complete_chat_async(
         """
         request_settings.messages = messages
         request_settings.stream = False
-        if request_settings.ai_model_id is None:
+        if request_settings.ai_model_id == '':
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:62]: why not use `if not request_settings.ai_model_id:`, with that both an empty string and a None value get handled.

Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:62]: The same thing needs to happen in `complete_chat_stream_async`!

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ async def complete_async(
         """
         request_settings.prompt = prompt
         request_settings.stream = False
-        if request_settings.ai_model_id is None:
+        if request_settings.ai_model_id == '':
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py:56]: same as before

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ async def complete_stream_async(
         """
         request_settings.prompt = prompt
         request_settings.stream = True
-        if request_settings.ai_model_id is None:
+        if request_settings.ai_model_id == '':
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py:82]: same as before

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
small syntactic change catches both cases (none and "") and the chat stream is missing.

**Comment by imfht** [comment]
> small syntactic change catches both cases (none and "") and the chat stream is missing.

thanks for your point. please see the latest commit.

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py [hunk 5]
```diff
@@ -31,6 +31,7 @@ class OllamaTextCompletion(TextCompletionClientBase, AIServiceClientBase):
         url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/generate
     """
 
+    ai_model_id: str = "llama2"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py:34]: do we need this and I would like to keep the annotations on the ai_model_id field from AIServiceClientBase: `ai_model_id: Annotated[str, StringConstraints(strip_whitespace=True, min_length=1)]`

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_embedding.py [hunk 6]
```diff
@@ -27,6 +27,7 @@ class OllamaTextEmbedding(EmbeddingGeneratorBase, AIServiceClientBase):
         session {Optional[aiohttp.ClientSession]} -- Optional client session to use for requests.
     """
 
+    ai_model_id: str = "llama2"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_embedding.py:30]: same as the other one, and does llama2 support embeddings?

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 7]
```diff
@@ -30,11 +30,12 @@ class OllamaChatCompletion(TextCompletionClientBase, ChatCompletionClientBase, A
     Make sure to have the ollama service running either locally or remotely.
 
     Arguments:
-        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library, defaults to llama2
         url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/chat
         session {Optional[aiohttp.ClientSession]} -- Optional client session to use for requests.
     """
 
+    ai_model_id: str = "llama2"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:38]: same as text_completion, keep annotation and why this default?

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
just a question on the default remaining!

**Comment by imfht** [comment]
> just a question on the default remaining!

Hi Eduard, thanks for your point, I updated my code now.
I'm new to semantic-kernel, If there is any questions, please feel free to point them out. 

wow~

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 8]
```diff
@@ -30,11 +30,12 @@ class OllamaChatCompletion(TextCompletionClientBase, ChatCompletionClientBase, A
     Make sure to have the ollama service running either locally or remotely.
 
     Arguments:
-        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        ai_model_id {str} -- Ollama model name, defaults to llama2 (the most popular model at https://ollama.ai/library)
         url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/chat
         session {Optional[aiohttp.ClientSession]} -- Optional client session to use for requests.
     """
 
+    ai_model_id: str = "llama2"
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:38]: this is really not necessary, instantiating without a `ai_model_id` is not possible anyway, and model popularity is due to change over time, so I would rather not put this in. The only needed change to fix the issue is the one below.

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py (hunk 8 above)
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:38]: Let's please remove this default `llama2` ai_model_id value in this file and the ollama text embedding file, and this should PR should be good to go. Thank you.

**Review by moonbox3 — CHANGES_REQUESTED** [review]
A small change to get this PR finished.

**Pull request closed without merging**